src/rules/add-income.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * @overview Add income attribute based on user's IP zipcode. 
+ * @gallery true
+ * @category enrich profile
+ * 
+ * Add zipcode median household income to the user profile.
+ * 
+ * This rule will add an `income` (median household income) attribute to the user based on the zipcode of their ip address.
+ * It is based on the last US Census data (not available for other countries).
+ * 
+ */
+
+function (user, context, callback) {
+
+  user.user_metadata = user.user_metadata || {};
+  const geoip = user.user_metadata.geoip || context.request.geoip;
+
+  if (!geoip || geoip.country_code !== 'US') return callback(null, user, context);
+
+  if(global.incomeData === undefined) {
+    retrieveIncomeData(user, geoip, context, callback);
+  } else {
+    setIncomeData(global.incomeData, user, geoip, context, callback);
+  }
+
+  function retrieveIncomeData(user, geoip, context, callback) {
+    request.get('http://cdn.auth0.com/zip-income/householdincome.json', function (e,r,b) {
```

Comment:
I know it was originally like this, but let's make these parameters a little more explanatory. The point of the template is to show users what's going on, so they're almost like documentation in that sense. We can leverage the request option for `json: true`, so the `JSON.parse` is unnecessary.
Also, let's clean up the spacing, so it's more consistent.

```
request.get('http://cdn.auth0.com/zip-income/householdincome.json', { json: true }, function (error, response, body) {
  if (error) return callback(error);
  if (response.statusCode === 200) {
    global.incomeData = body;
    setIncomeData(body, user, geoip, context, callback);
  }
  callback(null, user, context);
});
```

_note:_ this would also require a change in `add-income.test.js` to account for the extra param with `request` like:
` get: jest.fn((uri, opts, cb) => {`